lang/ghc/Portfile

```diff
@@ -29,144 +24,145 @@ long_description    \
-# errors. GHC _should_ actually compile itself in stage2 using paths from the
-# command line arguments
-compiler.cpath /usr/include
-compiler.library_path /usr/lib
-
-build.args      VERBOSE=1
-destroot.args   VERBOSE=1
-
-pre-activate {
-    set obsoletes [list]
-
-    # Legacy port deactivation hack added 2012-12-08, hs-platform-ghc no longer exists
-    lappend obsoletes hs-platform-ghc
-    # Legacy port deactivation hack added 2013-08-11, hs-process is provided by base
-    lappend obsoletes hs-process
-    # Legacy port deactivation hack added 2013-08-14, hs-template-haskell is provided by base
-    lappend obsoletes hs-template-haskell
-    # Legacy port deactivation hack added 2015-01-05, hs-transformers, hs-xhtml
-    # and hs-binary are provided by base
-    lappend obsoletes hs-transformers
-    lappend obsoletes hs-xhtml
-    lappend obsoletes hs-binary
-
-    foreach obsolete $obsoletes {
-        if {![catch {set obsolete_installed [lindex [registry_active $obsolete] 0]}]} {
-            # $obsolete is installed and active
-            # force deactivation
-            registry_deactivate_composite $obsolete "" [list ports_nodepcheck 1]
-        }
+supported_archs     x86_64
+
+depends_fetch-append \
+                    port:gnupg2
+
+post-checksum {
+    # check GPG signature: https://www.haskell.org/ghc/download_ghc_8_6_5.html
+    # pre-load public key to avoid keyserver downtime issues
+    # https://pgp.mit.edu/pks/lookup?op=get&search=0x2DE04D4E97DB64AD
+    # note: tcl exec will return error if error messages not directed to /dev/null
+    set gpg_keyid 97db64ad
+    system -W ${workpath} "/bin/sh -c '${prefix}/bin/gpg --homedir ${workpath}/.gnupg --import ${filespath}/keyid-${gpg_keyid}.txt 2>/dev/null || /usr/bin/true'"
+    set gpg_verification [exec /bin/sh -c "if ${prefix}/bin/gpg --homedir ${workpath}/.gnupg --verify ${distpath}/${name}-${version}-x86_64-apple-darwin${extract.suffix}.sig ${distpath}/${name}-${version}-x86_64-apple-darwin${extract.suffix} 2>/dev/null; then echo 'VERIFIED'; else echo 'UNVERIFIED'; fi"]
+    if {[string trim ${gpg_verification}] != "VERIFIED"} {
+        ui_error "GPG signature verification failed with keyid ${keyid}."
     }
 }
 
-post-activate {
-    set libprefix "${prefix}/lib/${name}-${version}"
-
-    # delete old ${prefix}/lib/ghc-${version} directories
-    foreach fullpath [glob -nocomplain -directory ${prefix}/lib ghc-*] {
-        if {$fullpath ne $libprefix} {
-            ui_info "Cleaning up remnant GHC library directory ${fullpath}"
-            catch {delete $fullpath}
+extract.only        ${name}-${version}-x86_64-apple-darwin${extract.suffix}
+
+# fix ${buildmakejobs} to be hw.physicalcpu+1 
+# https://github.com/macports/macports-base/blob/4b36e605d76fa3a423b4e35a6722ade4185ffa4f/src/port1.0/portbuild.tcl#L137
+namespace path { ::tcl::mathop }
+global buildmakejobs
+set buildmakejobs   [+ [sysctl hw.physicalcpu] 1]
```

Comment:
Okay.

Reply:
Nope that's necessary with the current approach.

See https://github.com/macports/macports-ports/pull/4794#issuecomment-512063225

Anyone who wants to build this will have to devote half their `hw.physicalcpu`'s. That's a lot better than the MacPorts default of all `hw.logicalcpu`'s.

Reply:
Now fixed with a call to `portbuild::build_getjobsarg`.